src/storage/compression/numeric_constant.cpp

```diff
@@ -0,0 +1,158 @@
+#include "duckdb/function/compression/compression.hpp"
+#include "duckdb/storage/buffer_manager.hpp"
+#include "duckdb/common/types/vector.hpp"
+#include "duckdb/storage/statistics/numeric_statistics.hpp"
+#include "duckdb/storage/statistics/validity_statistics.hpp"
+#include "duckdb/storage/table/column_segment.hpp"
+#include "duckdb/function/compression_function.hpp"
+#include "duckdb/storage/segment/uncompressed.hpp"
+
+namespace duckdb {
+
+//===--------------------------------------------------------------------===//
+// Scan
+//===--------------------------------------------------------------------===//
+unique_ptr<SegmentScanState> ConstantInitScan(ColumnSegment &segment) {
+	return nullptr;
+}
+
+//===--------------------------------------------------------------------===//
+// Scan base data
+//===--------------------------------------------------------------------===//
+void ConstantScanFunctionValidity(ColumnSegment &segment, ColumnScanState &state, idx_t scan_count, Vector &result) {
+	auto &validity = (ValidityStatistics &)*segment.stats.statistics;
+	if (validity.has_null) {
+		result.SetVectorType(VectorType::CONSTANT_VECTOR);
+		ConstantVector::SetNull(result, true);
+	}
+}
+
+template <class T>
+void ConstantScanFunction(ColumnSegment &segment, ColumnScanState &state, idx_t scan_count, Vector &result) {
+	auto &nstats = (NumericStatistics &)*segment.stats.statistics;
+
+	auto data = FlatVector::GetData<T>(result);
+	data[0] = nstats.min.GetValueUnsafe<T>();
+	result.SetVectorType(VectorType::CONSTANT_VECTOR);
+}
+
+//===--------------------------------------------------------------------===//
+// Scan Partial
+//===--------------------------------------------------------------------===//
+void FillFunctionValidity(ColumnSegment &segment, Vector &result, idx_t start_idx, idx_t count) {
```

Comment:
Little worried that these names don't have `Constant` in them - might cause collisions?

Reply:
Added